### PR TITLE
Add the ablitiy to create section introductions

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
This Pr modifies the build groups structure to allow publisher to publish surveys with a section introduction.

### How to review 
Tests pass and when tested against a survey which includes a introduction the schema is correct and displays the introduction as it should.

### CODE REVIEW

File: src/eq_schema/Block.js [hunk 1]
```diff
@@ -38,6 +38,15 @@ class Block {
...
 
+  static buildIntroBlock({ introductionTitle, introductionContent }, groupId) {
+    return {
+      type: "Interstitial",
+      id: `group${groupId}-introduction`,
+      title: introductionTitle || "",
+      description: introductionContent || ""
+    };
+  }
+
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply src/eq_schema/Block.js]: I mostly put it here as it felt neater and easier to find even though we do have to call it from elsewhere, and I had a feeling that soon there might be a bunch of static methods in here for all the different block types that are coming up, but if you think it's cleaner to move it up a level I can do?

[the remark this answers is not in the document]
Comment (SamGodwin2) [reply src/eq_schema/Block.js]: Is it worth creating a whole new class object just for the small introduction block, feels like that's a bit overkill. Not sure there's any happy middle ground here without adding complexity where it's not necessary. 

I'd be happy to bring it up a level, in my mind it feels like it should be with the block methods, but I definitely see where you're coming from in that there's no real reason for it to be a static method on another class?

File: src/eq_schema/Group.js [hunk 2]
```diff
@@ -34,8 +34,12 @@ class Group {
     );
   }
 
-  buildBlocks(pages, groupId, ctx) {
-    return pages.map(page => new Block(page, groupId, ctx));
+  buildBlocks(pages, groupId, introduction, ctx) {
+    const blocks = pages.map(page => new Block(page, groupId, ctx));
+    if (introduction.introductionEnabled) {
+      return concat(Block.buildIntroBlock(introduction, groupId), blocks);
```
Comment (tgandrews) [inline src/eq_schema/Group.js]:
```suggestion
      return blocks.concat(Block.buildIntroBlock(introduction, groupId));
```

But we'd also need to remove the import.

Comment (SamGodwin2) [reply src/eq_schema/Group.js]: Wouldn't this put the intro blocks after the blocks array though? And you can't do it the other way as buildIntroBlock doesn't return an array?

Comment (tgandrews) [reply src/eq_schema/Group.js]:
```suggestion
      return [Block.buildIntroBlock(introduction, groupId), ...blocks];
```

Good point. I missed that!

**Pull request merged**